includes/wikia/models/PromoImage.class.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * Helper class for handling images used for promotion
+ */
+class PromoImage extends WikiaObject {
+	const MAIN_IMAGE_BASE_NAME = 'Wikia-Visualization-Main';
+	const ADDITIONAL_IMAGES_BASE_NAME = 'Wikia-Visualization-Add';
+	const IMAGES_EXT = '.png';
+
+	const INVALID = -1;
+	const MAIN = 0;
+	const ADDITIONAL_START = 1;
+	const ADDITIONAL_END = 9;
+
+	private $dbName, $cityId, $type, $fileChanged, $removed;
+
+	static public function fromPathname($pathString){
+		$type = self::inferType($pathString, $dbName);
+		return new PromoImage($type, $dbName);
+	}
+
+	public function __construct($type, $dbName = null) {
+		parent::__construct();
+		$this->dbName = $dbName;
+		$this->cityId = null;
+		$this->type = $type;
+		$this->fileChanged = false;
+		$this->removed = false;
+	}
+
+	private function isType($type){
+		return $this->type === $type;
+	}
+
+	private function getDBName(){
+		if (empty($this->dbName) && !empty($this->cityId)) {
+			$this->dbName = WikiFactory::IDtoDB($this->cityId);
+		}
+		return $this->dbName;
+	}
+
+	private function isCityIdSet(){
+		return (!empty($this->dbName) || !empty($this->cityId));
+	}
+
+	public function setCityId($cityId) {
+		$this->cityId = $cityId;
+		$this->dbName = null;
+		return $this;
+	}
+
+	private function getCityId(){
+		if (empty($this->cityId) && !empty($this->dbName)){
+			$this->cityId = WikiFactory::DBtoID($this->dbName);
+		}
+		return $this->cityId;
+	}
+
+	//only saves city id if it is not set
+	public function ensureCityIdIsSet($cityId){
+		if (!$this->isCityIdSet()) {
+			$this->setCityId($cityId);
+		}
+		return $this;
+	}
+
+	private function pathnameHelper($withDbName = true, $withExtension = true){
+		if ($this->isType(self::INVALID)){
+			$path = null;
+		} else {
+			if ($this->isType(self::MAIN)){
+				$path = self::MAIN_IMAGE_BASE_NAME;
+			} else {
+				$path = self::ADDITIONAL_IMAGES_BASE_NAME . "-" . $this->type;
+			}
+			if ($withDbName && $this->isCityIdSet()) {
+				$path .= ',' . $this->getDBName();
+			}
+			if ($withExtension) {
+				$path .= self::IMAGES_EXT;
+			}
+		}
+		return $path;
+	}
+
+	public function getPathname(){
+		return $this->pathnameHelper(true, true);
+	}
+
+	public function getOriginFile($cityId = null){
+		if (empty($cityId)){
+			$cityId =  $this->getCityId();
+		}
+
+		$f = GlobalFile::newFromText($this->getPathname(), $cityId);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$f". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)